src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -99,10 +107,116 @@ private static bool TryDecodeSuppressMessageAttributeData (CustomAttribute attri
+			}
+		}
+
+		private void DecodeGlobalSuppressMessageAttributes (ICustomAttributeProvider provider, GlobalSuppressions globalSuppressions)
+		{
+			var attributes = provider.CustomAttributes.Where (a => a.AttributeType.Name == "UnconditionalSuppressMessageAttribute");
+			foreach (var instance in attributes) {
+				SuppressMessageInfo info;
+				if (!TryDecodeSuppressMessageAttributeData (instance, out info))
+					continue;
+
+				ModuleDefinition module = null;
+				if (provider is ModuleDefinition _module)
+					module = _module;
+				else if (provider is AssemblyDefinition assembly)
+					module = assembly.MainModule;
+				else if (provider is TypeDefinition type)
+					module = type.Module;
+				else if (provider is IMemberDefinition member)
+					module = member.DeclaringType.Module;
+				else {
+					_context.LogMessage (MessageContainer.CreateInfoMessage ("`UnconditionalSuppressMessage` attribute was placed in an language element which is currently not supported."));
+				}
+
+				Debug.Assert (module != null);
+				if (info.Target == null && (info.Scope == "module" || info.Scope == null)) {
+					AddLocalSuppression (instance, module);
+					continue;
+				}
+
+				if (info.Target == null || info.Scope == null)
+					continue;
+
+				switch (info.Scope) {
+				case "type":
+				case "member":
+					foreach (var result in DocumentationSignatureParser.GetMembersForDocumentationSignature (info.Target, module))
+						globalSuppressions.AddGlobalSuppression (result, info);
+
+					break;
+				case "resource":
+				case "module":
+				case "namespace":
+				case "namespaceanddescendants":
+					_context.LogMessage (MessageContainer.CreateInfoMessage ($"Scope `{info.Scope}` used in `UnconditionalSuppressMessage` is currently not supported."));
+					break;
+				default:
+					break;
+				}
+			}
+		}
+
+		private class GlobalSuppressions
+		{
+			private readonly Dictionary<ICustomAttributeProvider, Dictionary<int, SuppressMessageInfo>> _globalSuppressions = new Dictionary<ICustomAttributeProvider, Dictionary<int, SuppressMessageInfo>> ();
+
+			public bool HasGlobalSuppressions (int id, ICustomAttributeProvider provider, out SuppressMessageInfo info)
+			{
+				Dictionary<int, SuppressMessageInfo> suppressions = null;
+				if (_globalSuppressions.TryGetValue (provider, out suppressions) && suppressions != null)
+					return suppressions.TryGetValue (id, out info);
+
+				info = default;
+				return false;
+			}
+
+			public void AddGlobalSuppression (ICustomAttributeProvider provider, SuppressMessageInfo info)
+			{
+				if (_globalSuppressions.TryGetValue (provider, out var suppressions)) {
+					AddOrUpdate (info, suppressions);
+					return;
+				}
+
+				var _suppressions = new Dictionary<int, SuppressMessageInfo> { { info.Id, info } };
+				_globalSuppressions.Add (provider, _suppressions);
+			}
+
+			private static void AddOrUpdate (SuppressMessageInfo info, IDictionary<int, SuppressMessageInfo> builder)
```

Comment:
The name says Add, where is that code?

Reply:
We're adding or updating by using the indexer. If we use the dictionary `Add` method here, we could be throwing an exception. I can change the name of this method it it's misleading.

Reply:
I don't think that is what the code does. It would if you removed `ContainsKey` logic